api/envoy/extensions/filters/http/dynamic_forward_proxy/v3/dynamic_forward_proxy.proto

```diff
@@ -25,19 +25,21 @@ message FilterConfig {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.config.filter.http.dynamic_forward_proxy.v2alpha.FilterConfig";
 
-  // The DNS cache configuration that the filter will attach to. Note this configuration must
-  // match that of associated :ref:`dynamic forward proxy cluster configuration
-  // <envoy_v3_api_field_extensions.clusters.dynamic_forward_proxy.v3.ClusterConfig.dns_cache_config>`.
-  common.dynamic_forward_proxy.v3.DnsCacheConfig dns_cache_config = 1;
+  oneof implementation_specifier {
+    // The DNS cache configuration that the filter will attach to. Note this configuration must
+    // match that of associated :ref:`dynamic forward proxy cluster configuration
+    // <envoy_v3_api_field_extensions.clusters.dynamic_forward_proxy.v3.ClusterConfig.dns_cache_config>`.
+    common.dynamic_forward_proxy.v3.DnsCacheConfig dns_cache_config = 1;
+
+    // The timeout used for sub cluster initialization. Defaults to 5s if not set.
+    google.protobuf.Duration cluster_init_timeout = 3 [(validate.rules).duration = {gt {}}];
```

Comment:
Let's move this into a dedicated SubClusterConfig message so we can add more configuration in the future, thanks.

/wait